src/common/http/httpc.cpp

```diff
@@ -187,8 +187,17 @@ optional<Error> HttpClient::send_request(const char *host, Connection *conn, Req
     }
 
     static const constexpr HeaderOut term = { nullptr, nullptr, nullopt };
+    static constexpr size_t buff_size { 16 }; // 4294967295 (10 digits) is the max number that fits into size_t
+    char buff[buff_size];
     for (const HeaderOut *extra_hdrs = request.extra_headers() ?: &term; extra_hdrs->name; extra_hdrs++) {
-        CHECKED(buffer.header(extra_hdrs->name, extra_hdrs->value, extra_hdrs->size_limit));
+        if (std::holds_alternative<const char *>(extra_hdrs->value)) {
+            CHECKED(buffer.header(extra_hdrs->name, std::get<const char *>(extra_hdrs->value), extra_hdrs->size_limit));
+        } else if (std::holds_alternative<size_t>(extra_hdrs->value)) {
+            snprintf(buff, buff_size, "%u", std::get<size_t>(extra_hdrs->value));
+            CHECKED(buffer.header(extra_hdrs->name, buff, extra_hdrs->size_limit));
+        } else {
+            return Error::InternalError;
+        }
```

Comment:
Can we use some kind of `std::visit` that guarantees all variants are handled in constant time?

Reply:
Continued in https://github.com/prusa3d/Prusa-Firmware-Buddy/pull/2556